src/lib/util/RunOutputProcessor.ts

```diff
@@ -28,9 +28,25 @@ export class RunOutputProcessor {
 
 
 	public processRunOutput(rrr: RecombinedRuleResults): AnyJson {
-		const {minSev, results} = rrr;
+		const {minSev, results, summaryMap} = rrr;
+
+		const hasViolations = [...summaryMap.values()].some(summary => summary.violationCount !== 0);
+
+		// If there are neither violations nor an outfile, then we want to avoid writing empty
+		// results to the console.
+		// NOTE: If there's an outfile, we skip this part. This is because we still want to generate
+		//       an empty outfile
+		if (!this.opts.outfile && !hasViolations) {
```

Comment:
Before #1170, we were using this `if` to log a message and return early if there were no violations.
#1170 removed this block entirely.
To restore the original behavior for the non-`--outfile` case, we've restored the block, but modified the `if` condition so it's only satisfied when there's no violations _and_ no outfile.